File: src/main/java/io/koju/autopos/trade/sale/service/SaleService.java
package io.koju.autopos.trade.sale.service;

import io.koju.autopos.trade.sale.domain.Sale;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.transaction.annotation.Transactional;

public interface SaleService {

    @Transactional(readOnly = true)
    Page<Sale> getSalesWithStatus(Sale.Status status, Pageable pageable);
}

File: src/main/scala/io/koju/autopos/trade/sale/service/SaleService.scala
package io.koju.autopos.trade.sale.service

import io.koju.autopos.catalog.domain.SaleStatus
import io.koju.autopos.trade.sale.domain.{Sale, SaleLine}
import org.springframework.data.domain.{Page, Pageable}

trait SaleService {

  def getSalesWithStatus(status: SaleStatus, pageable: Pageable): Page[Sale]

  def createNewSale(): Sale

  def updateSale(sale: Sale): Sale

  def addSaleLine(sale: Sale, saleLine: SaleLine): SaleLine

  def updateSaleLine(sale: Sale, saleLine: SaleLine): SaleLine

  def deleteSaleLine(saleLine: SaleLine): Unit

}
